### DISCUSSION TITLE
Using Django with Wave

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [h2oai/wave#934] [Q&A]
Hello,
I have used PyWebIO for a while, now got informed about Wave, and would like to confess, it's very neat library.
some questions tough:

can we use wave inside Django ? 
the reason of my question is following:

- handling user authentication with Django,
- having a blog which can be easily maintained.
- using other databases than wavedb, for example Mongodb.
- be able to do authorization. 

My question can be stupid, as I have background in data science and not WebDev.

Thanks in advance

**Comment by Shahin-rmz** [comment]
I'm still eager to find any possible answeres if there is, thanks.

**Reply by lo5** [reply]
The answer depends on what you want to achieve. Wave supports open ID connect for authentication and authorization. You can use any database you want. Whatever works in regular Python should work as is in Wave, as long as the libraries you use support asyncio.

**Comment by ebdavison** [comment]
I am also interested in this. We have been working on dashboards and this provides all we need and more. But we have already built a rather large Django app/site and would like to put wave into one of the “pages” to replace the somewhat meager dashboard already built. Is there a way to maybe embed this? Or load it in a similar way to how a react “app” would be called from an HTML page?

**Comment by lo5** [comment]
@ebdavison @Shahin-rmz Maybe Nitro (https://nitro.h2o.ai/) is a better fit. Nitro is designed to be plugged in into existing frameworks. We have examples for how to integrate with Flask, Tornado, and Starlette/Uvicorn. A Django sample is missing, but should be straightforward with a suitable websocket handler. 

Note that this is a fairly new library, so it's marked as alpha, but it's already used in production use at H2O.
